pulsar-io/core/pom.xml

```diff
@@ -38,16 +38,10 @@
 
     <dependency>
       <groupId>${project.groupId}</groupId>
-      <artifactId>pulsar-common</artifactId>
+      <artifactId>pulsar-functions-api</artifactId>
       <version>${project.version}</version>
-      <exclusions>
```

Comment:
pulsar-common brings many different dependencies which doesn't require for pulsar-io. so, I think we should exclude all transitive dependencies unless it actually requires.

Reply:
Actually, the diff is misleading, but the change is correct: 
 * Removed dep on `pulsar-common`
 * Added dep on `pulsar-functions-api` since record is moved there